src/os_win/os_errno.c

```diff
@@ -63,14 +63,33 @@ int
 __wt_errno(void)
 {
 	/*
+	 * Check for 0:
+	 * It's easy to introduce a problem by calling the wrong error function,
+	 * for example, this function when the MSVC function set the C runtime
+	 * error value. Handle gracefully and always return an error.
+	 */
+	return (errno == 0 ? WT_ERROR : errno);
```

Comment:
> There's an additional change, to always return an error value

I wasn't clear - that is the change I was concerned with. Maybe we could add the assertions back in to both error handlers (and clear the errno in ``__wt_errno``), so we can detect if something has gone wrong in diagnostic builds.